doc/examples/segment_extending_clustering_framework.py

```diff
@@ -0,0 +1,233 @@
+colormap_full = np.ones((len(streamlines), 3))
+for cluster, color in zip(clusters, colormap):
+    colormap_full[cluster.indices] = color
+
+ren = fvtk.ren()
+ren.SetBackground(1, 1, 1)
+fvtk.add(ren, fvtk.streamtube(streamlines, colormap_full))
+fvtk.record(ren, n_frames=1, out_path='fornix_clusters_arclength.png', size=(600, 600))
+
+"""
+.. figure:: fornix_clusters_arclength.png
+   :align: center
+
+   **Showing the different clusters obtained by using the arc length**.
+
+
+Extending `Metric`
+==================
+This section will guide you through the creation of a new metric that can be
+used in the context of this clustering framework.
+
+Assuming a set of streamlines, we want a metric that computes the cosine
+distance giving the vector between endpoints of each streamline (i.e. one
+minus the cosine of the angle between two vectors). For more information
+about this distance check `<http://en.wikipedia.org/wiki/Cosine_similarity>`_.
+
+Let's start by importing the necessary modules.
+"""
+
+from dipy.segment.metric import Metric
+from dipy.segment.metric import VectorBetweenEndpointsFeature
+
+"""
+We now define the class `CosineMetric` that will perform the desired
+distance computation. When subclassing `Metric`, two methods have to be
+redefined: `are_compatible` and `dist`. Moreover, when implementing the
+`dist` method, one needs to make sure the distance returned is symmetric
+(i.e. `dist(A, B) == dist(B, A)`).
+"""
+
+
+class CosineMetric(Metric):
+    """ Computes the cosine distance between two streamlines. """
+    def __init__(self):
+        # For simplicity, features will be the vector between endpoints of a streamline.
+        super(CosineMetric, self).__init__(feature=VectorBetweenEndpointsFeature())
+
+    def are_compatible(self, shape1, shape2):
+        """ Checks if two features are vectors of same dimension.
+
+        Basically this method exists so we don't have to do this check
+        inside the `dist` method (speedup).
+        """
+        return shape1 == shape2 and shape1[0] == 1
+
+    def dist(self, v1, v2):
+        """ Computes a the cosine distance between two vectors. """
+        norm = lambda x: np.sqrt(np.sum(x**2))
+        cos_theta = np.dot(v1, v2.T) / (norm(v1)*norm(v2))
+
+        # Make sure it's in [-1, 1], i.e. within domain of arccosine
+        cos_theta = np.minimum(cos_theta, 1.)
+        cos_theta = np.maximum(cos_theta, -1.)
+        return np.arccos(cos_theta) / np.pi  # Normalized cosine distance
+
+"""
+The new distance `CosineMetric` is ready to be used. Let's use
+it to cluster a set of streamlines according the cosine distance of the
+vector between their endpoints. For educational purposes we will try to
+cluster a small streamline bundle known from neuroanatomy as the fornix.
+
+We start by loading the fornix streamlines.
+"""
+
+import numpy as np
+from nibabel import trackvis as tv
+from dipy.data import get_data
+from dipy.viz import fvtk
+
+fname = get_data('fornix')
```

Comment:
If you want to use a different bundle than the fornix. I have some simple cingulum bundles (cb) in the data. Look in get_data.
Will try to have some more bundles asap.